Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/ProcessWeightTests.cs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using BuildXL.Scheduler.Distribution;
+using Test.BuildXL.Executables.TestProcess;
+using Test.BuildXL.Scheduler;
+using Test.BuildXL.TestUtilities.Xunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace IntegrationTest.BuildXL.Scheduler
+{
+    public class ProcessWeightTests : SchedulerIntegrationTestBase
+    {
+        public ProcessWeightTests(ITestOutputHelper output) : base(output)
+        {
+            // Turns on logging for most scheduler stats, including the limiting resource stat looked for by these tests
+            ShouldLogSchedulerStats = true;
+        }
+
+        [Theory]
+        [InlineData(-1)] // Make sure a weights of < 1 are treated the same as a weight of 1
+        [InlineData(0)]
+        [InlineData(1)]
+        public void CanRunUpToMaxProcessesInParallel(int defaultWeight)
+        {
+            CreateAndScheduleProcessWithWeight(defaultWeight, numProcesses: Configuration.Schedule.MaxProcesses);
+
+            RunScheduler().AssertSuccess();
+            AssertProcessConcurrencyWeightLimited(false);
+        }
+
+        [Theory]
+        [InlineData(-1)] // Make sure a weights of < 1 are treated the same as a weight of 1
+        [InlineData(0)]
+        [InlineData(1)]
+        public void CannotRunGreaterThanMaxProcessesInParallel(int defaultWeight)
+        {
+            // Schedule one over max concurrent processes
+            CreateAndScheduleProcessWithWeight(defaultWeight, numProcesses: Configuration.Schedule.MaxProcesses + 1);
+
+            RunScheduler().AssertSuccess();
+            AssertProcessConcurrencyWeightLimited(true);
+        }
+
+        [Fact]
+        public void UseProcessWeightToRunAlone()

```

Comment:
>UseProcessWeightToRunAlone [](start = 20, length = 26)

From jus the AssertProcessConcurrency...(...), I find it hard to distinguish this test from `CannotRunGreaterThanMaxProcessesInParallel`. Could we have a better assertion?

Reply:
Can you suggest an assertion you prefer?

This test checks that I can specify a "solo" pip by using the weight feature. 

CannotRunGreaterThanMaxProcessesInParallel is a sanity check that we have a limit to the amount of processes you can run in paralle.

---
In reply to: [271098853](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271098853) [](ancestors = 271098853)

Reply:
Resolving after offline discussion.

---
In reply to: [271100350](https://github.com/Microsoft/BuildXL/pull/67#discussion_r271100350) [](ancestors = 271100350,271098853)